src/test/java/org/gridsuite/filter/server/FilterEntityControllerTest.java

```diff
@@ -315,4 +433,98 @@ private StringBuilder numericalRange(String id, RangeType range, Double value1,
+        }
+        filter += "}";
+
+        insertFilter(id, filter);
+
+        List<FilterAttributes> filterAttributes = objectMapper.readValue(
+            mvc.perform(get("/" + FilterApi.API_VERSION + "/metadata").contentType(APPLICATION_JSON).content("[\"" + id + "\"]"))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString(),
+            new TypeReference<>() {
+            });
+
+        assertEquals(1, filterAttributes.size());
+        assertEquals(name, filterAttributes.get(0).getName());
+        assertEquals(id, filterAttributes.get(0).getId());
+        assertEquals(type, filterAttributes.get(0).getType());
+        assertEquals(description, filterAttributes.get(0).getDescription());
+
+        mvc.perform(delete(URL_TEMPLATE + id)).andExpect(status().isOk());
+    }
+
+    private void insertTransformerFilter(FilterType type, String name, UUID id, String equipmentID, String equipmentName,
+                                       String substationName, String description, Set<String> countries,
+                                       List<RangeType> rangeTypes, List<Double> values1, List<Double> values2)  throws Exception {
+        String filter = "{" + joinWithComma(
+            jsonVal("name", name),
+            jsonVal("id", id.toString()),
+            jsonVal("type", type.name()));
+
+        if (equipmentID != null) {
+            filter += ", " + jsonVal("equipmentID", equipmentID);
+        }
+        if (equipmentName != null) {
+            filter += ", " + jsonVal("equipmentName", equipmentName);
+        }
+        if (substationName != null) {
+            filter += ", " + jsonVal("substationName", substationName);
+        }
+        if (description != null) {
+            filter += ", " + jsonVal("description", description);
+        }
+        if (rangeTypes != null) {
+            for (int i = 0; i < rangeTypes.size(); ++i) {
+                filter += ", " + numericalRange("nominalVoltage" + (i + 1), rangeTypes.get(i), values1.get(i), values2.get(i));
+            }
+        }
+        if (countries != null) {
+            filter += ", " + jsonSet("countries", countries);
+        }
+        filter += "}";
+
+        insertFilter(id, filter);
+
+        List<FilterAttributes> filterAttributes = objectMapper.readValue(
+            mvc.perform(get("/" + FilterApi.API_VERSION + "/metadata").contentType(APPLICATION_JSON).content("[\"" + id + "\"]"))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString(),
+            new TypeReference<>() {
+            });
+
+        assertEquals(1, filterAttributes.size());
+        assertEquals(name, filterAttributes.get(0).getName());
+        assertEquals(id, filterAttributes.get(0).getId());
+        assertEquals(type, filterAttributes.get(0).getType());
+        assertEquals(description, filterAttributes.get(0).getDescription());
```

Comment:
Why not check nominalVoltage and countries ?

Reply:
It is already tested in the insertFilter method called from this method